uaclient/api/api.py

```diff
@@ -0,0 +1,76 @@
+from importlib import import_module
+from typing import Callable, List, Optional
+
+from uaclient.api.data_types import APIData, APIResponse
+from uaclient.api.errors import APIError, error_out
+from uaclient.data_types import DataObject
+
+VALID_ENDPOINTS = ["u.pro.version.v1"]
+
+
+def call_api(endpoint_path: str, options: List[str]) -> APIResponse:
```

Comment:
Sorry I didn't notice this earlier. I think `options` should be type `Dict[str, Any]` and the `split("=")` should happen in the cli action.

@lucasmoura maybe we can just make this adjustment in your magic attach pr to keep this one moving though?

Reply:
This is an implementation detail that isn't used in this PR so I don't think it should block this PR

Reply:
Good point - It's an implementation detail we can always change later, don't worry about it